modules/codegen/src/smithy4s/codegen/Renderer.scala

```diff
@@ -634,36 +669,36 @@ private[codegen] class Renderer(compilationUnit: CompilationUnit) { self =>
       renderHintsVal(hints),
       newline,
       values.map { case e @ EnumValue(value, ordinal, _, _) =>
-        line"""case object ${e.name} extends $name("$value", "${e.name}", $ordinal)"""
+        line"""case object ${TypeReference(
```

Comment:
I think we could have shorthand methods on the [Decl](https://github.com/disneystreaming/smithy4s/blob/main/modules/codegen/src/smithy4s/codegen/IR.scala#L35) type : 

```
def nameDef = ??? 
def nameRef = ??? 
```

that'd make the code clearer I think

Reply:
EnumValue and Operations  are  not of type Decl . 
We can add it in and cover unions,products , and type aliases

Reply:
yup sure